airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -502,7 +497,11 @@ def generate_scd_type_2_model(self, from_table: str, column_names: Dict[str, Tup
 
     def get_cursor_field(self, column_names: Dict[str, Tuple[str, str]]) -> str:
         if self.cursor_field and len(self.cursor_field) == 1:
-            return column_names[self.cursor_field[0]][0]
+            if not is_airbyte_column(self.cursor_field[0]):
```

Comment:
what happened here?

Reply:
`column_names` is a map of all original columns declared in the properties/JSON schema for the current table.

But, here we want to allow generated airbyte columns as cursor fields or primary key too (those may not be listed in the original JSON schema) so I add that special case with this `if` statement